qt: Remove QFont warnings with QT_QPA_PLATFORM=minimal

hebasto opened the pull request:
This PR removes massive warnings like:
```
QWARN  : ... QFont::setPointSizeF: Point size <= 0 (...), must be greater than 0
```

from `test_bitcoin-qt` output.

On master (e258ce792a4849927a6db51786732d71cbbb65fc):
```
$ ./src/qt/test/test_bitcoin-qt | grep QFont | wc -l
~BitcoinApplication : Stopping thread
~BitcoinApplication : Stopped thread
57
```

With this PR:
```
$ ./src/qt/test/test_bitcoin-qt | grep QFont | wc -l
~BitcoinApplication : Stopping thread
~BitcoinApplication : Stopped thread
0
```

hebasto committed:
qt: Remove QFont warnings with QPA=minimal

practicalswift commented:
ACK 8ab99dd9544e6fa61da1cd77922c85e2970c0416 -- tests should be silence in the absence of errors: thanks for fixing! :)

promag reviewed (commented):
Not sure about this change. What if this is eventually fixed on Qt? Are these warnings worth the change?

Alternatively you could filter the warning in `DebugMessageHandler` or even call `qInstallMessageHandler` with a specific test message handler.

Either way a comment explaining the condition would be nice.

hebasto commented:
Updated 8ab99dd9544e6fa61da1cd77922c85e2970c0416 -> d318a14cc87f672198aea7490d76950919357f7d ([pr17908.01](https://github.com/hebasto/bitcoin/commits/pr17908.01) -> [pr17908.02](https://github.com/hebasto/bitcoin/commits/pr17908.02), [compare](https://github.com/hebasto/bitcoin/compare/pr17908.01..pr17908.02)).

@promag  
> Not sure about this change. What if this is eventually fixed on Qt? Are these warnings worth the change?

Tests should not be noisy, IMO. 57 warnings messages look spammy ;)

> Either way a comment explaining the condition would be nice.

A comment has been added.

MarcoFalke commented on src/qt/rpcconsole.cpp:
```diff
@@ -496,7 +497,7 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
     ui->peerHeading->setText(tr("Select a peer to view detailed information."));
 
-    consoleFontSize = settings.value(fontSizeSettingsKey, QFontInfo(QFont()).pointSize()).toInt();
+    consoleFontSize = settings.value(fontSizeSettingsKey, QFont().pointSize()).toInt();
```
Why is this change needed? And what is the observable change?

https://doc.qt.io/qt-5/qfontinfo.html#details

MarcoFalke replied:
I see, but when QT_QPA_PLATFORM is set to something else, we actually want the values that are used to draw the text, not the point size of `QFont{}`, no?

hebasto replied:
It seems there is no harm when `RPCConsole::setFontSize()` is called. With non-`qminimal` plugin Qt always could draw a correct font.

MarcoFalke replied:
I am not an expert in qt, and it might work for you. But in the general case it might depend on the fonts that are selected in the system settings.

hebasto replied:
[Qt Documentation -> Drawing and Filling](https://doc.qt.io/qt-5/paintsystem.html):
> Qt will use the font with the specified attributes, or if no matching font exists, Qt will use the closest matching installed font.

promag commented on src/qt/transactionview.cpp:
```diff
@@ -396,8 +397,9 @@ void TransactionView::contextualMenu(const QPoint &point)
+    if (index.isValid() && QGuiApplication::platformName() != "minimal") {
+        // The qminimal plugin does not provide window system integration.
+        // Skip this call to suppress warnings during the tests with QT_QPA_PLATFORM=minimal.
         contextMenu->popup(transactionView->viewport()->mapToGlobal(point));
```
If you keep this approach maybe refactor this to something like `GUIUtil::popup(contextMenu, ...)` and remove the new include which is already in guiutil.cpp.

hebasto replied:
[Done](https://github.com/bitcoin/bitcoin/pull/17908#issuecomment-599260885).

promag reviewed (commented):
> Alternatively you could filter the warning in `DebugMessageHandler` or even call `qInstallMessageHandler` with a specific test message handler.

@hebasto have you considered this approach?

hebasto commented:
@promag 
> > Alternatively you could filter the warning in `DebugMessageHandler` or even call `qInstallMessageHandler` with a specific test message handler.
> 
> @hebasto have you considered this approach?

- it seems `DebugMessageHandler` is not used in `test_bitcoin-qt`

- `qInstallMessageHandler` with a specific test message handler could suppress warnings for possible new cases in the future

hebasto commented:
Updated d318a14cc87f672198aea7490d76950919357f7d -> 50c2791e35929c53c351e82a6188e2b58f5151e8 ([pr17908.02](https://github.com/hebasto/bitcoin/commits/pr17908.02) -> [pr17908.03](https://github.com/hebasto/bitcoin/commits/pr17908.03), [diff](https://github.com/hebasto/bitcoin/compare/pr17908.02..pr17908.03)):

- implemented @promag's [suggestion](https://github.com/bitcoin/bitcoin/pull/17908#discussion_r392632929)

promag reviewed (commented):
Code review ACK 50c2791e35929c53c351e82a6188e2b58f5151e8.

hebasto commented on src/qt/guiutil.cpp:
```diff
@@ -904,4 +904,11 @@ void LogQtInfo()
+void QMenuPopup(QMenu* menu, const QPoint& point, QAction* atAction)
+{
+    // The qminimal plugin does not provide window system integration.
+    if (QGuiApplication::platformName() == "minimal") return;
```
Actually, `QApplication` inherits `platformName()` from `QGuiApplication` :tiger2:

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/17908#issuecomment-619684824).

hebasto commented:
Updated 50c2791e35929c53c351e82a6188e2b58f5151e8 -> 2a388bdf8f6685d639713c903443458afc2458e2 ([pr17908.03](https://github.com/hebasto/bitcoin/commits/pr17908.03) -> [pr17908.04](https://github.com/hebasto/bitcoin/commits/pr17908.04), [diff](https://github.com/hebasto/bitcoin/compare/pr17908.03..pr17908.04)):

- addressed @promag's comments

hebasto commented on src/qt/guiutil.cpp:
```diff
@@ -904,4 +904,11 @@ void LogQtInfo()
     }
 }
 
+void QMenuPopup(QMenu* menu, const QPoint& point, QAction* atAction)
```
[Updated](https://github.com/bitcoin/bitcoin/pull/17908#issuecomment-619684824).

promag commented:
Code review ACK 2a388bdf8f6685d639713c903443458afc2458e2.

hebasto commented:
Rebased 2a388bdf8f6685d639713c903443458afc2458e2 -> 1122817c194ed49abf896e68604e725c3b5c8569 ([pr17908.04](https://github.com/hebasto/bitcoin/commits/pr17908.04) -> [pr17908.05](https://github.com/hebasto/bitcoin/commits/pr17908.05)) due to the conflict with #16710.

promag commented:
Code review ACK 1122817c194ed49abf896e68604e725c3b5c8569.

Only Qt issue I found that could be related is https://bugreports.qt.io/browse/QTBUG-72335.

jonasschnelli commented:
utACK 1122817c194ed49abf896e68604e725c3b5c8569
